File: 3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i/3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i.java
class Solution {
    public int minOperations(int[] nums) {
        int count = 0;
        int i = 0;
        while (i < nums.length) {
            if (nums[i] == 0) {
                if (i + 2 < nums.length) {
                    for (int j = i; j < i + 3; ++j) {
                        nums[j] = 1 - nums[j];
                    }
                    count++;
                } else {
                    return -1;
                }
            }
            i++;
        }
        return count;
    }
}

File: 3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i/3191-minimum-operations-to-make-binary-array-elements-equal-to-one-i.cpp
class Solution {
public:
    int minOperations(vector<int>& nums) {
      
        int count=0,n=nums.size();
        
        for(int i=0;i<n;i++){
            if(nums[i]==0){
               if(i+3>n){
                   return -1;
               }
               for(int j=i;j<i+3;j++){
                   nums[j]=!nums[j];
               } 
                count++;
            }
        }
        return count;
    }
};